### PR TITLE
Write logs according the verbosity level defined

### PR DESCRIPTION
**Pull request by erickhun**
This PR will :

- allow to change verbosity by setting the `LOG_LEVEL` in the env variable
- "silent" all logs lower than the `NOTICE` level

I use [Monolog levels](https://github.com/Seldaek/monolog/blob/1.25.2/src/Monolog/Logger.php#L95-L104) for now until we decide the final ones 

@colinscape @esclapes Can I have your eyes on this ?

### CODE REVIEW

File: src/BuffLog/BuffLog.php [hunk 1]
```diff
@@ -47,27 +55,50 @@ public static function __callStatic($methodName, $args)
...
             }
         } else {
             error_log("BuffLog::$methodName() does not exist");
         }
     }
 
+    private static function setVerbosityLevel()
+    {
+        $logLevelFromEnv = getenv(self::$logLevelEnvVar);
+        $monologLevels = self::getLogger()->getLevels();
```
Comment (esclapes) [inline src/BuffLog/BuffLog.php]: We could slide this two lines out and pass `$logLevel` to the `setVerbosityLevel` as an argument. That way this method becomes a setter that could be used to configure the verbosity level in _user land._ I don't think that's going to happen very often, though.

File: src/BuffLog/BuffLog.php [hunk 2]
```diff
@@ -47,27 +55,50 @@ public static function __callStatic($methodName, $args)
...
+                error_log("LOG_LEVEL {$logLevelFromEnv} is not defined, use one of: " . implode(', ', array_keys($monologLevels)));
+            }
+        }
+    }
+
+
+    private static function shouldWriteLogs($methodName)
+    {
+        // Optimization: we could move this at the singleton initialization
+        self::setVerbosityLevel();
```
Comment (esclapes) [inline src/BuffLog/BuffLog.php]: :plus-one: to that optimization comment :+1:

**Review by esclapes — APPROVED** [review]
Hey @erickhun, thanks for this work :taco: 

I think monolog already supports this feature. If I am not mistaken we could configure it by passing it as the second parameter to the `StreamHandler` constructor:

```php
$logLevelFromEnv = getenv(self::$logLevelEnvVar);
$handler = new StreamHandler('php://stdout', $logLevelFromEnv);
```

I may be missing some context on the requirements as there may be good reasons to build our own. If that's the case, this implementation LGTM :+1: I left a couple of small comments below. Nothing blocking!

**Comment by erickhun** [comment]
@esclapes oh wow , great spot, I didn't spot that feature 🙈 This will greatly simplify the code 🤩. Let me rework the code and loop you again with this

**Comment by erickhun** [comment]
@esclapes I've reworked the code  with using the `StreamHandler` feature for verbosity . Definitely cleaner (and better!) 🌮 🌮

**Review by colinscape — APPROVED** [review]
Looks good to me! 👍

**Review by esclapes — APPROVED** [review]
Great that it helped @erickhun LGTM! :taco: :+1:

**Pull request merged**